ansible-scylla-node/tasks/Debian.yml

```diff
@@ -1,6 +1,19 @@
 ---
 - name: Add Scylla repos
   block:
+  - name: "Purge keyring '{{ scylla_repo_keyringfile }}'"
+    ansible.builtin.file:
+      path: "{{ scylla_repo_keyringfile }}"
+      state: absent
+    when: install_type == 'online' and scylla_repo_keyserver is defined and scylla_repo_keys is defined and (scylla_repo_keys|length > 0)
+
+  - name: Remove an apt key by id previously added
```

Comment:
You need to move both these tasks to be after the one that installs `gnupg2` (line 17 below).